doc/tutorials/enable_s5.rst

```diff
@@ -31,131 +31,197 @@ The diagram below shows the overall architecture:
-shut down successfully or not, and to decide when to shut the Service VM itself down.
-
-User VM "Lifecycle Manager"
-===========================
-
-As part of the S5 reference design, a Lifecycle Manager daemon (``life_mngr`` in Linux,
-``life_mngr_win.exe`` in Windows) runs in the User VM to implement S5. It waits for the shutdown
-request from the Service VM on the serial port. The simple protocol between the Service VM and
-User VM is as follows: when the daemon receives ``shutdown``, it sends ``ACKED`` to the Service VM;
-then it shuts down the User VM. If the User VM is not ready to shut down,
-it can ignore the ``shutdown`` command.
+When a human operator initiates the flow through running ``s5_trigger_linux.py`` or ``s5_trigger_win.py``,
+the Lifecycle Manager (``life_mngr``) running in that User VM will send the system S5 request via
+the vUART to the Lifecycle Manager in the Service VM which in turn acknowledges the request.
+The Lifecycle Manager in Service VM will send ``poweroff`` request to User VMs, when the Lifecycle Manager
+in User VMs receives ``poweroff`` request, it sends ``ack_poweroff`` to the Service VM;
+then it shuts down the User VMs. If the User VMs is not ready to shut down, it can ignore the
+``poweroff`` request.
+
+.. note:: The User VM need to be authorized to be able to request a system S5, this is achieved
+   by configuring  ``ALLOW_TRIGGER_S5`` in the Lifecycle Manager service configuration file ``/etc/life_mngr.conf``
+   in the Service VM. There is only one User VM in the system can be configured to request a shutdown.
+   If this configuration is wrong, the system S5 request from User VM will be rejected by
+   Lifecycle Manager of Service VM, the following error message will be recorded in Lifecycle Manager
+   log file ``/var/log/life_mngr.log`` of Service VM:
+   ``The user VM is not allowed to trigger system shutdown``
+
+Initiate a System S5 within the Service VM
+==========================================
+
+On the Service VM side, it uses the ``s5_trigger_linux.py`` to trigger the system S5 flow. Then,
+the Lifecycle Manager in service VM sends a ``poweroff`` request to the lifecycle manager in each
+User VM through the vUART channel. If the User VM receives this request, it will send an ``ack_poweroff``
+to the lifecycle manager in Service VM. It is the Service VM's responsibility to check whether the
+User VMs shut down successfully or not, and to decide when to shut the Service VM itself down.
+
+.. note:: Service VM is always allowed to trigger system S5 by default.
 
 .. _enable_s5:
 
 Enable S5
 *********
 
-The procedure for enabling S5 is specific to the particular OS:
-
-* For Linux (LaaG) or Windows (WaaG), include these lines in the launch script:
-
-  .. code-block:: bash
+1. Configure communication vUART for Service VM and User VMs:
+
+   Add these lines in the hypervisor scenario XML file manually:
```

Comment:
Is this really the only option for now? We want to avoid as much as possible that users edit these XML files directly.

Reply:
Thanks! Good question. This manual XML file modification is only needed in v2.7 due to config tool GUI limitation. If config tool updates are done, this manual modification can be removed.

Reply:
@nanlin @junjiemao1 We need to keep the configurator updated for these new options in v2.7, right?

Reply:
@dbkinder , Since configurator is ongoing redesign, thus we would like to hold the big change for configurator until we have new design implementation.